### PR TITLE
feat: Add Sound Browser to REPL with comprehensive UX improvements

### PR DESCRIPTION
**Pull request by samasastudio**
## Summary

Adds a comprehensive Sound Browser feature to the REPL, allowing users to explore, preview, and discover 400+ sample categories from the Strudel Dirt Samples library. This feature integrates seamlessly into the REPL as a collapsible bottom tray with intelligent window resizing.

### Key Features

**Core Functionality:**
- Fetches and caches 400+ sample categories from Strudel Dirt Samples GitHub repository
- Real-time search filtering across all categories
- 13 organized instrument groups (Drums, Bass, Kicks, Snares, Cymbals, Percussion, etc.)
- Audio preview with play/stop controls
- Displays samples in Strudel pattern format (e.g., "808:0")

**UI/UX Improvements:**
- Music icon toggle button in REPL header (replaces Test button when engine is ready)
- Responsive CSS grid layout (2-5 columns based on screen size) for easy browsing
- Smart window auto-resize: expands by 434px when browser opens, with smooth 300ms animations
- Sound browser and editor share space equally (both flex-1)
- Keyboard navigation: Escape key stops preview
- Group tabs with icons for quick filtering
- Persistent state via localStorage (open/closed, selected group)

**Performance & Polish:**
- Module-level caching in SampleRegistry prevents duplicate fetches
- React Strict Mode compatible (proper effect cleanup and cancellation tracking)
- Smooth animations without dragging lag (transitions only on width/height, not position)
- Glassmorphic styling consistent with Basilisk AV design system
- Loading and error states with user feedback

### Architecture

**New Services:**
- `sampleRegistry.ts` - Centralized sample metadata fetching and caching

**New Components:**
- `SoundBrowserTray` - Main container with search, group tabs, categories, and samples
- `SoundSearch` - Real-time search input with debouncing
- `SoundCategoryChips` - Horizontal scrollable category selector with counts
- `SoundSampleGrid` - Responsive grid of sample buttons with play/stop

**New Hooks:**
- `useSampleData` - Fetches and manages sample metadata with proper cancellation
- `useSoundPreview` - Handles audio preview with Strudel evaluate/hush
- `useSoundBrowser` - Orchestrates browser state (open, group, category, search)

**Enhanced Components:**
- `REPLWindow` - Lifts sound browser state for proper layout control
- `StrudelRepl` - Receives sound browser as prop, handles Esc key behavior
- `useREPLWindow` - Implements derived state pattern with useMemo for smart resize

**New Configuration:**
- `sampleGroups.ts` - Keyword-based categorization for 13 instrument groups

### Test Coverage

- **74 new tests added** (224 total, 223 passing)
- `sampleRegistry.test.ts` - Service layer tests
- `useSampleData.test.ts` - Hook tests with loading/error states
- `useSoundPreview.test.ts` - Audio preview tests
- `StrudelRepl.test.tsx` - Updated component tests

### Files Changed

19 files with 1,786 insertions, 29 deletions:
- 4 new UI components
- 3 new hooks
- 1 new service
- 1 new config file
- 2 enhanced components (REPLWindow, StrudelRepl)
- 1 enhanced hook (useREPLWindow)
- 4 new test files
- 1 updated test file

### Manual Testing

- [ ] **Open/Close Browser**
  - Click Music icon in REPL header
  - Verify tray opens with smooth 300ms animation
  - Verify REPL window expands by ~434px automatically
  - Click Music icon again to close
  - Verify window returns to original size smoothly
  
- [ ] **Browse Samples**
  - Open sound browser and verify 13 group tabs appear (All, Drums, Bass, Kicks, etc.)
  - Click through different groups and verify categories update
  - Click category chips and verify samples appear in responsive grid (2-5 columns)
  - Verify grid layout shows rows and columns, not single horizontal line
  - Verify samples display in Strudel format (e.g., "808:0", "bd:1")

- [ ] **Audio Preview**
  - Select a category and click a sample's play button
  - Verify audio plays and button shows stop icon (filled square)
  - Verify sample button highlights with cyan accent color
  - Click stop button to halt preview
  - Click different sample while one is playing - verify first stops, new one plays
  
- [ ] **Search Functionality**
  - Type in search box (e.g., "kick")
  - Verify categories filter in real-time
  - Verify "No categories found matching..." message when no results
  - Clear search and verify all categories return

- [ ] **Keyboard Navigation**
  - Open sound browser and preview a sample
  - Press Escape key
  - Verify audio stops but browser stays open
  - Preview another sample and press Escape again to confirm behavior

- [ ] **Window Interactions**
  - Drag REPL window around with sound browser open
  - Verify dragging is smooth with no lag or flickering
  - Resize REPL window with sound browser open
  - Verify both editor and browser remain visible and usable
  - Verify min-height of 300px for sound browser

- [ ] **State Persistence**
  - Open sound browser and select a specific group
  - Close and reopen browser
  - Verify it reopens with same group selected
  - Refresh page and verify browser state persists via localStorage

- [ ] **Loading & Error States**
  - Open sound browser on slow connection
  - Verify "Loading samples..." message appears
  - Verify categories load without infinite spinner (React Strict Mode fix)
  - Test with network disconnected to verify error handling

### Automated Testing

- [ ] Run `npm test` and verify all 224 tests pass (1 unrelated flaky test acceptable)
- [ ] Verify no TypeScript errors: `npm run build`
- [ ] Verify no lint errors: `npm run lint`

### Edge Cases

- [ ] Very small viewport - verify min-width/height constraints work
- [ ] Very large viewport - verify max-width/height (90vw/vh) respected
- [ ] Select category with only 1-2 samples - verify grid layout still looks good
- [ ] Search with special characters - verify no errors
- [ ] Rapidly toggle browser open/close - verify animations don't break

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/StrudelRepl.tsx [hunk 1]
```diff
@@ -105,11 +109,24 @@ type Props = {
     onExecute?: () => void;
     onSave?: (code: string) => void;
     statusLabel?: string;
+    soundBrowser: UseSoundBrowserReturn;
 };
 
-export const StrudelRepl = ({ className, engineReady, onTestPattern, onHalt, onExecute, onSave, statusLabel }: Props): JSX.Element => {
+export const StrudelRepl = ({ className, engineReady, onTestPattern: _onTestPattern, onHalt, onExecute, onSave, statusLabel, soundBrowser }: Props): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (samasastudio) [reply src/components/StrudelRepl.tsx]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/samasastudio/basilisk-av/pull/20#discussion_r2615158121)

File: src/App.tsx [hunk 2]
```diff
@@ -12,6 +13,18 @@ import { focusREPL } from './utils/focusREPL';
...
+    queries: {
+      staleTime: Infinity, // Data never becomes stale (like our cache)
+      gcTime: Infinity, // Keep in cache forever (was cacheTime in v4)
+      retry: 3,
+      refetchOnWindowFocus: false,
+    },
+  },
+});
+
 export const App = (): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (samasastudio) [reply src/App.tsx]: @copilot lets go ahead with this change

File: src/components/sound-browser/SoundSampleGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return (
+          <button
+            key={sampleKey}
+            onClick={() => {
+              if (isPlaying && onStopPreview) {
+                onStopPreview();
+              } else {
+                onPreviewSample(categoryName, index);
+              }
+            }}
```
[the remark this answers is not in the document]
Comment (samasastudio) [reply src/components/sound-browser/SoundSampleGrid.tsx]: @copilot lets go ahead with this change
